### PR TITLE
#139456249  setting up test enviroment and integrating coverall code coverage service

### PR DESCRIPTION
**Pull request by freemile2ru**
# **What does this PR do?**

    Setup unit testing for the front end
    Setup coverall code coverage service

# **Description of Task to be completed?**
   1. this PR sets jasmine framework to test our codes and also configures karma to make firefox, chrome and phantom js our testing environment
   2. it also configures karma to send our test coverage result to travis using covaralls.  

# **How should this be manually tested?**

## **for the unit testing**
(1)clone this repository
(2)create the .js file you want to test and save.
(3) enter the test folder
(4) create new .js file you want to write your test in
(5) include the scripts path in the specRunner.html file with the actual code path coming before the test code path
(6)load the specRunner.html file in your browser
   

## For test coverage service
 (1) clone the repo,
 (2) install all dependencies
 (3) navigate to the location on your Cli
 (4) run karma start karma2.conf.js
 (5) the browsers will be launched to show karma is working and the tests results on each browser will be displayed on the console
 (6) the test coverage result will be shown the coverage folder and on this repository when pushed.

# **Any background context you want to provide?**
  this PR contains a travis.yml file for your test coverage

# **What are the relevant pivotal tracker stories?**
    Setup unit testing for the front end
    Setup coverall code coverage service
    
Screenshots (if appropriate)
Questions:

### CODE REVIEW

File: lib/jasmine-2.3.4/boot.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/**
+ Starting with version 2.0, this file "boots" Jasmine, performing all of the necessary initialization before executing the loaded environment and all of a project's specs. This file should be loaded after `jasmine.js` and `jasmine_html.js`, but before any project source files or spec files are loaded. Thus this file can also be used to customize Jasmine for a project.
```
Comment (ooduye) [inline lib/jasmine-2.3.4/boot.js]: Remove the lib/ directory from the pull request

File: config/env/all.js [hunk 2]
```diff
@@ -5,5 +5,5 @@ var keys = rootPath + '/keys.txt';
 module.exports = {
 	root: rootPath,
 	port: process.env.PORT || 3000,
-    db: process.env.MONGOHQ_URL
+    db: `mongodb://freemile:Kratus043@ds151279.mlab.com:51279/card4human`
```
Comment (ooduye) [inline config/env/all.js]: Change this to reflect just the environment variable

**Comment by ooduye** [comment]
Your travis-ci build is failing. Please check it and find out why.

**Comment by freemile2ru** [comment]
hello oluwayemisi,
  i have fixed the bug and travis is now building. thanks

**Comment by ooduye** [comment]
@andela-uibrahim Your builds are failing...

**Comment by freemile2ru** [comment]
hello @andela-ooduye,
 I have implemented the feedback you requested. i have changed the file names, change the coverage  file format and the travis is now building. please help review the current status

**Review by ooduye — APPROVED** [review]

**Pull request merged**